### ISSUE TITLE
(symbolic) dsolve should return expression instead of equation

### ISSUE THREAD

**Issue by mass2010chromium** [cbm755/octsympy#1003]
When I execute the following code in matlab, it works without any problems:
```
clear all
clc
syms y(t)
dy = diff(y)
deq = diff(y, 2) + dy + y == 0
soln = dsolve(deq, y(0) == 1, dy(0) ==  0)
y_eval = double(vpa(subs(soln, t, 4)))
```
and gets out a double value. But running this code in octave breaks because `dsolve` returns an Equality, which gets processed by `vpa()` into another equality, which can't be converted to a single number by `double()`.

![octave](https://user-images.githubusercontent.com/11793417/73709508-ebb1fa00-46c6-11ea-9feb-8edd0c613ba0.png)
Notice how in the second to last line in the image, the result is type sym instead of an expression / just a vpa number.

A simple fix would be to force dsolve to return the rhs of the value it gets from sympy's dsolve, but this might break some things that rely on multiple return values from dsolve or implicit solutions.

I don't know what matlab does if it can't find an explicit solution, more testing needed.

TL;DR `dsolve()` should return an expression for compatibility with matlab.

**Comment by mtmiller** [comment]
Thanks for the detailed bug report!

Just for some more info, according to the docs for [Matlab `dsolve`](https://www.mathworks.com/help/symbolic/dsolve.html), the return value should be analogous to `solve`. If solving one differential equation, return a scalar symbolic expression. If solving a system of differential equations, return a struct with fields named after the functions, or as multiple return values if `nargout` is greater than 1.

**Comment by mass2010chromium** [comment]
Do we want octave's `dsolve` to behave just like matlab's `dsolve`, and not like sympy's `dsolve`? Because right now there's some code in dsolve that is doing classifying that definitely isn't in the matlab spec, and it gets in the way of returning multiple solution equations.

**Comment by mass2010chromium** [comment]
PR: #1004

**Comment by cbm755** [comment]
It would be nice if it behaved more like Matlab, but we also want to minimize how much code we're maintaining (to be different from SymPy).  Can be hard to balance those things.

I have try to review your merge request when I get a chance: thanks very much for that.
